### PR TITLE
Tidies up the configuration to better support VS Code

### PR DESCRIPTION
**Pull request by huntc**
This configuration update reflects the strategy taken with Embassy where a `Cargo.example.toml` is provided. This example is expected to be updated by the developer and customised for the target they are interested in. Without these changes, working with `nrf-hal` is quite unpleasant in Visual Studio Code.

A `Cargo.ci.toml` is provided and used by CI. This file is the original `Cargo.toml`.

Thanks to @Dirbaio for the recommendations.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,6 +37,12 @@ Every nRF chip has its own crate, listed below:
 
 \* These devices do not have a separate development kit and share the [NRF52 DK](https://www.nordicsemi.com/Software-and-tools/Development-Kits/nRF52-DK)
 
+## Development
+
+Be sure to copy and edit the `Cargo.example.toml` file to `Cargo.toml`. The file will require editing dependent on the target you wish to work with and contains some further
+instructions. Similarly, check out the `.vscode/settings.json` file when used in the context of Visual Studio Code. By default, all of theses files are configured to work
```
Comment (jonas-schievink) [inline README.md:43]:
```suggestion
instructions. Similarly, check out the `.vscode/settings.json` file when used in the context of Visual Studio Code. By default, all of these files are configured to work
```

**Review by jonas-schievink — APPROVED** [review]
Not too happy about this hack, but having rust-analyzer support is more important, so :+1:

**Comment by jonas-schievink** [comment]
Hmm, this still doesn't make rust-analyzer work properly with the provided example file. It seems to think that the `51`, `52810`, `52811`, `52832`, `52833`, `52840` and `9160` features are all enabled at the same time.

**Comment by huntc** [comment]
> Hmm, this still doesn't make rust-analyzer work properly with the provided example file. It seems to think that the `51`, `52810`, `52811`, `52832`, `52833`, `52840` and `9160` features are all enabled at the same time.

I shall be investigate.

**Comment by huntc** [comment]
> Hmm, this still doesn't make rust-analyzer work properly with the provided example file. It seems to think that the `51`, `52810`, `52811`, `52832`, `52833`, `52840` and `9160` features are all enabled at the same time.

If I take the `Cargo.example.toml` as `Cargo.toml` then open `nvmc.rs` from common, I see this:

image

i.e. the correct highlighting was given that I'm just bringing in the 52840 feature. Do you see something similar?

If so then how are you observing that all features are enabled?

**Comment by jonas-schievink** [comment]
Ah, found the issue, I had `rust-analyzer.cargo.allFeatures` enabled globally, will add that to our VS Code configuration.

**Pull request merged**